x/eibc/keeper/msg_server.go

```diff
@@ -23,25 +26,25 @@ var _ types.MsgServer = msgServer{}
 func (m msgServer) FulfillOrder(goCtx context.Context, msg *types.MsgFulfillOrder) (*types.MsgFulfillOrderResponse, error) {
 	ctx := sdk.UnwrapSDKContext(goCtx)
 	logger := ctx.Logger()
-	// Check that the msg is valid
+
 	err := msg.ValidateBasic()
 	if err != nil {
 		return nil, err
 	}
-	// Check that the order exists in status PENDING
-	demandOrder, err := m.GetDemandOrder(ctx, commontypes.Status_PENDING, msg.OrderId)
+
+	demandOrder, err := m.GetActiveOrder(ctx, msg.OrderId)
 	if err != nil {
 		return nil, err
 	}
-	// Check that the order is not fulfilled yet
-	if demandOrder.IsFulfilled {
-		return nil, types.ErrDemandAlreadyFulfilled
-	}
 
-	// Check the underlying packet is still relevant (i.e not expired, rejected, reverted)
-	if demandOrder.TrackingPacketStatus != commontypes.Status_PENDING {
-		return nil, types.ErrDemandOrderInactive
+	// Check that the fulfiller expected fee is equal to the demand order fee
+	for _, coin := range demandOrder.Fee {
+		expectedFee, _ := sdk.NewIntFromString(msg.ExpectedFee)
+		if !coin.Amount.Equal(expectedFee) {
+			return nil, types.ErrExpectedFeeNotMet
+		}
 	}
+
 	// Check for blocked address
 	if m.bk.BlockedAddr(demandOrder.GetRecipientBech32Address()) {
 		return nil, types.ErrBlockedAddress
```

Comment:
I know it's not your PR but why is this here, rather than before demand order creation?

Reply:
yeah seems redundant